### PR TITLE
Add _SINGLE_PROCESS property to CachedDataSet

### PR DESCRIPTION
**Pull request by carlaprv**
## Description
Solves https://github.com/kedro-org/kedro/issues/1888

## Development notes
The CachedDataSet cannot be used with the ParellelRunner this PR adds the `_SINGLE_PROCESS` property just like in [DeltaTableDataSet](https://github.com/kedro-org/kedro/blob/main/kedro/extras/datasets/spark/deltatable_dataset.py#L60-L64)

Before, this PR trying to use CachedDataSet and ParallelRunner together was failing.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1905"></a>

### CODE REVIEW

**Comment by carlaprv** [comment]
Linting is failing for a file that I didn't modify 
image
image

File: kedro/io/cached_dataset.py [hunk 1]
```diff
@@ -27,6 +27,12 @@ class CachedDataSet(AbstractDataSet):
     class as shown above.
     """
 
+    # this dataset cannot be used with ``ParallelRunner``,
+    # therefore it has the attribute ``_SINGLE_PROCESS = True``
+    # for parallelism within a Spark pipeline please consider
+    # ``ThreadRunner`` instead
```
Comment (MerelTheisenQB) [inline kedro/io/cached_dataset.py:33]: We can remove these two sentences, because for the `CachedDataSet` this is not related to Spark in any way.

**Review by MerelTheisenQB — COMMENTED** [review]
Thank you for your contribution! 😄 I've fixed the linter in another branch, so it should pass now. 

I've left one minor suggestion and then it can be merged!

**Review by jmholzer — APPROVED** [review]
Congratulations on your first PR! 🎉 Great work.

For future reference, working on a branch on kedro-org/kedro repo is totally fine and is the way we normally do it. It simplifies the workflow by quite a bit 🙂.

File: kedro/io/cached_dataset.py (hunk 1 above)
Comment (carlaprv) [reply kedro/io/cached_dataset.py:33]: Sure! Thanks for the suggestion

**Comment by carlaprv** [comment]
> Congratulations on your first PR! 🎉 Great work.
> 
> For future reference, working on a branch on kedro-org/kedro repo is totally fine and is the way we normally do it. It simplifies the workflow by quite a bit 🙂.

Thanks @jmholzer! I've followed the process in the [contribution guidelines](https://kedro.readthedocs.io/en/stable/contribution/developer_contributor_guidelines.html). Next time, I'll create a branch direct on `kedro` repo.

File: kedro/io/cached_dataset.py (hunk 1 above)
Comment (carlaprv) [reply kedro/io/cached_dataset.py:33]: @MerelTheisenQB may we could keep the suggestion to use ``ThreadRunner``?

````
# for parallelism please consider ``ThreadRunner`` instead
````

**Comment by jmholzer** [comment]
> > Congratulations on your first PR! 🎉 Great work.
> > For future reference, working on a branch on kedro-org/kedro repo is totally fine and is the way we normally do it. It simplifies the workflow by quite a bit 🙂.
> 
> Thanks @jmholzer! I've followed the process in the [contribution guidelines](https://kedro.readthedocs.io/en/stable/contribution/developer_contributor_guidelines.html). Next time, I'll create a branch direct on `kedro` repo.

Ohh I see 😃 I did the same for my first PR. Thanks for reminding us, let me see about creating an issue to update our contributor guidelines.

**Review by MerelTheisenQB — APPROVED** [review]
Thanks again for the contribution! ⭐ ⭐ ⭐
